src/compiler/transformers/esnext.ts

```diff
@@ -16,9 +20,157 @@ namespace ts {
+            switch (node.kind) {
+                case SyntaxKind.ParenthesizedExpression: return visitNonOptionalParenthesizedExpression(node as ParenthesizedExpression, captureThisArg);
+                case SyntaxKind.PropertyAccessExpression: return visitNonOptionalPropertyAccessExpression(node as PropertyAccessExpression, captureThisArg);
+                case SyntaxKind.ElementAccessExpression: return visitNonOptionalElementAccessExpression(node as ElementAccessExpression, captureThisArg);
+                case SyntaxKind.CallExpression: return visitNonOptionalCallExpression(node as CallExpression, captureThisArg);
+                default: return visitNode(node, visitor, isExpression);
+            }
+        }
+
+        function visitOptionalExpression(node: OptionalChain, captureThisArg: boolean): Expression {
+            const { expression, chain } = flattenChain(node);
+            const left = visitNonOptionalExpression(expression, isCallChain(chain[0]));
+            const temp = createTempVariable(hoistVariableDeclaration);
+            const leftThisArg = isSyntheticReference(left) ? left.thisArg : undefined;
+            const leftExpression = isSyntheticReference(left) ? left.expression : left;
+            let rightExpression: Expression = temp;
+            let thisArg: Expression | undefined;
+            for (let i = 0; i < chain.length; i++) {
+                const segment = chain[i];
+                switch (segment.kind) {
+                    case SyntaxKind.PropertyAccessExpression:
+                        if (i === chain.length - 1 && captureThisArg) {
+                            thisArg = createTempVariable(hoistVariableDeclaration);
+                            rightExpression = createParen(createAssignment(thisArg, rightExpression));
+                        }
+                        rightExpression = createPropertyAccess(
+                            rightExpression,
+                            visitNode(segment.name, visitor, isIdentifier)
+                        );
+                        break;
+                    case SyntaxKind.ElementAccessExpression:
+                        if (i === chain.length - 1 && captureThisArg) {
+                            thisArg = createTempVariable(hoistVariableDeclaration);
+                            rightExpression = createParen(createAssignment(thisArg, rightExpression));
+                        }
+                        rightExpression = createElementAccess(
+                            rightExpression,
+                            visitNode(segment.argumentExpression, visitor, isExpression)
+                        );
+                        break;
+                    case SyntaxKind.CallExpression:
+                        if (i === 0 && leftThisArg) {
+                            rightExpression = createFunctionCall(
+                                rightExpression,
+                                leftThisArg,

```

Comment:
1. why is leftThisArg the type argument here?
2. why is typeArguments undefined in the else clause? If it's always undefined, why?

Reply:
Yep, I got the two confused!